### DISCUSSION TITLE
Microservices

### DISCUSSION THREAD

**Discussion by Roman-Ch-32** [strawberry-graphql/strawberry#2803] [Q&A]
Вопрос, как наладить взаимодействие между сервисами. Проблема в том, что мне нужно перенаправить запрос на другой сервер, получить ответ и привести его к стравберри схемам. 
Каким образом в Сервисе предусмотрена возможно создавать микросервисы?

**сервисы написаны на разных языка**х

URL = "http://192.168.50.30:3002/graphql"
query = """query {sys_users {
                            id  
                            name
                            roles{
                            id
                            name
                            }  
                            avatar}}"""

async def get_user() -> list[Manager]:
    async with httpx.AsyncClient() as client:
        req = await client.request(method='post', url=URL, json={'query': query},
                                   headers={"allow_origins": "allow_origins", "access-control-allow-origin": "*"})
        return [Manager(**i) for i in req.json()['data']['sys_users']]

unpacking with ** does not allow you to look into the depths of the answer. And the schemes accept only the stove keyword arguments. At the same time, it is also unrealistic to transfer the list to another resolver, because this is prohibited by the service. Also, if you use another resolver in the schema field, the Resolver is initiated at an incomprehensible moment. Also, the scheme ceases to pay attention to the field in which the resolver is specified
@strawberry.type
class UserRole:
    id: int
    name: str

  @strawberry.type
  class Manager:
      id: strawberry.ID | None = None
      name: str | None = None
      avatar: str | None = None
      surname: str | None = None
      patronymic: str | None = None
      email: str | None = None
      roles: List[UserRole] = strawberry.field(resolver=resolver())

**Comment by Roman-Ch-32** [comment]
graph Company
@strawberry.federation.type(keys=["id", "managerId"], extend=True)
class Company:
id: strawberry.ID | None = None
company_name: str | None = None
company_full_name: str | None = None
info_type_buisness: Optional[CompanyBuisnesType] | None = None
autos: list[CompanyAuto] | None = None
autos_in_company: int | None = None
contacts: list[CompanyKontact] | None = None
inn: Optional[str]
manager_id: int | None = strawberry.federation.field(external=True)

graph Orders
@strawberry.federation.type(keys=["id", "managerId"])
class Company:
id: strawberry.ID | None = None
manager_id: int | None = strawberry.federation.field()
orders: List["Order"] = strawberry.field(resolver=get_order)

@classmethod
async def resolve_reference(cls, id: strawberry.ID, info: Info, manager_id: int, **kwargs) -> "Company":
    print(info.variable_values)
    return Company(id=id, manager_id=2)
{'representations': [{'id': '1', 'info': GraphQLResolveInfo(field_name='_entities', field_nodes=[FieldNode at 53:168], return_type=<GraphQLNonNull <GraphQLList <GraphQLUnionType '_Entity'>>>, parent_type=<GraphQLObjectType 'Query'>, path=Path(prev=None, key='_entities', typename='Query'), schema=<graphql.type.schema.GraphQLSchema object at 0x7fc35bbdd720>, fragments={}, root_value=None, operation=OperationDefinitionNode at 0:169, variable_values={...}, context={'request': <starlette.requests.Request object at 0x7fc35b90db40>, 'background_tasks': <starlette.background.BackgroundTasks object at 0x7fc35b90dc60>, 'response': <starlette.responses.Response object at 0x7fc35b90db70>}, is_awaitable=<function is_awaitable at 0x7fc35d8cc3a0>)}, {'id': '14', 'info': GraphQLResolveInfo(field_name='_entities', field_nodes=[FieldNode at 53:168], return_type=<GraphQLNonNull <GraphQLList <GraphQLUnionType '_Entity'>>>, parent_type=<GraphQLObjectType 'Query'>, path=Path(prev=None, key='_entities', typename='Query'), schema=<graphql.type.schema.GraphQLSchema object at 0x7fc35bbdd720>, fragments={}, root_value=None, operation=OperationDefinitionNode at 0:169, variable_values={...}, context={'request': <starlette.requests.Request object at 0x7fc35b90db40>, 'background_tasks': <starlette.background.BackgroundTasks object at 0x7fc35b90dc60>, 'response': <starlette.responses.Response object at 0x7fc35b90db70>}, is_awaitable=<function is_awaitable at 0x7fc35d8cc3a0>)}, {'id': '19', 'info': GraphQLResolveInfo(field_name='_entities', field_nodes=[FieldNode at 53:168], return_type=<GraphQLNonNull <GraphQLList <GraphQLUnionType '_Entity'>>>, parent_type=<GraphQLObjectType 'Query'>, path=Path(prev=None, key='_entities', typename='Query'), schema=<graphql.type.schema.GraphQLSchema object at 0x7fc35bbdd720>, fragments={}, root_value=None, operation=OperationDefinitionNode at 0:169, variable_values={...}, context={'request': <starlette.requests.Request object at 0x7fc35b90db40>, 'background_tasks': <starlette.background.BackgroundTasks object at 0x7fc35b90dc60>, 'response': <starlette.responses.Response object at 0x7fc35b90db70>}, is_awaitable=<function is_awaitable at 0x7fc35d8cc3a0>)}, {'id': '20', 'info': GraphQLResolveInfo(field_name='_entities', field_nodes=[FieldNode at 53:168], return_type=<GraphQLNonNull <GraphQLList <GraphQLUnionType '_Entity'>>>, parent_type=<GraphQLObjectType 'Query'>, path=Path(prev=None, key='_entities', typename='Query'), schema=<graphql.type.schema.<starlette.responses.Response object at 0x7fc35b90db70>}, is_awaitable=<function is_awaitable at 0x7fc35d8cc3a0>)}, {'id': '24', 'info': GraphQLResolveInfo(field_name='_entities', field_nodes=[FieldNode at 53:168], return_type=<GraphQLNonNull <GraphQLList <GraphQLUnionType '_Entity'>>>, parent_type=<GraphQLObjectType 'Query'>, path=Path(prev=None, key='_entities', typename='Query'), schema=<graphql.type.schema.GraphQLSchema object at 0x7fc35bbdd720>, fragments={}, root_value=None, operation=OperationDefinitionNode at 0:169, variable_values={...}, context={'request': <starlette.requests.Request object at 0x7fc35b90db40>, 'background_tasks': <starlette.background.BackgroundTasks object at 0x7fc35b90dc60>, 'response': <starlette.responses.Response object at 0x7fc35b90db70>}, is_awaitable=<function is_awaitable at 0x7fc35d8cc3a0>)}]}

only one field and only the ID is passed through the keys. an attempt to pass other fields causes the error " "Unable to resolve reference for <class 'query_schemas.Order'>""

**Comment by baseplate-admin** [comment]
please format the codes. At current stage it looks horrible ( and it probably reduces your chance of getting help )
